vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/view/boutiquesParsing/BoutiquesUtil.java

```diff
@@ -0,0 +1,245 @@
+     *
+     * @param descriptor    JSONObject to parse
+     * @param key           String representing the key in descriptor associated to searched value
+     * @param optional      boolean: true if key is optional, in which case its absence will lead to a null return
+     *                      value instead of a RuntimeException
+     * @return              Array of Strings associated to key in descriptor, or null if key is absent and optional is
+     *                      true
+     * @throws RuntimeException if expected value is not a valid String array or if key is absent and optional is false
+     */
+    public static String[] getArrayValueAsStrings(JSONObject descriptor, String key, boolean optional)
+            throws RuntimeException {
+        JSONArray array = getArrayValue(descriptor, key, optional);
+        try {
+            return array == null ? null : JSONArrayToStrings(array);
+        } catch(RuntimeException exception){
+            throw new RuntimeException("Invalid Boutiques descriptor: '" + key
+                    + "' array contains non-String element(s).");
+        }
+    }
+
+    /**
+     * Convert a JSONArray to a String array
+     *
+     * @param array JSONArray to convert
+     * @return      Array of Strings representing array
+     * @throws RuntimeException if some elements of array are not valid Strings
+     */
+    public static String[] JSONArrayToStrings(JSONArray array) throws RuntimeException {
+        String[] stringArray = new String[array.size()];
+        for (int valueNo = 0; valueNo < array.size(); valueNo++){
+            JSONString valueString = array.get(valueNo).isString();
+            if(valueString == null){
+                throw new RuntimeException("Invalid Array: value " + valueNo + " is not a String.");
+            }
+            stringArray[valueNo] = valueString.stringValue();
+        }
+        return stringArray;
+    }
+
+    /**
+     * Get a Map representing the JSON object associated to given key in given parent JSON object. Obtained JSON
+     * object must have String keys and String arrays as values
+     *
+     * @param descriptor    Parent JSONObject to parse
+     * @param key           String representing the key in descriptor associated to searched value
+     * @param optional      boolean: true if key is optional, in which case its absence will lead to a null return
+     *                      value instead of a RuntimeException
+     * @return              Map representing JSON object associated to key in descriptor, or null if key is absent and
+     *                      optional is true. Map's keys are obtained JSON object keys and values are String arrays
+     *                      representing its values
+     * @throws RuntimeException if expected value is not a valid object or if key is absent and optional is false
+     */
+    public static Map<String, String[]> getStringMapValue(JSONObject descriptor, String key,
```

Comment:
Why not. I guess it should also be Lists rather than Arrays, so `getStringListMap`? Maybe should it simply be `getMapValue` as it is the only case where we parse to return a map